backend/file-system/src/main/java/com/htec/filesystem/controller/ShelfController.java

```diff
@@ -77,4 +77,12 @@ public ShelfController(ShelfService shelfService) {
 
         return ResponseEntity.ok(shelfService.getFirstLevelTrash(authUser.getId()));
     }
+
+    @GetMapping("/check/{shelfId}")
```

Comment:
thing that you might choose more intuitive name (path) for access rights check endpoint? Because "check" does not tell much about the purpose of it.